x-pack/plugins/security_solution/public/entity_analytics/components/entity_store/hooks/use_entity_store.ts

```diff
@@ -5,150 +5,108 @@
+export const useEnableEntityStoreMutation = (options?: UseMutationOptions<{}>) => {
+  const { telemetry } = useKibana().services;
   const queryClient = useQueryClient();
+  const { enableEntityStore } = useEntityStoreRoutes();
 
-  return useCallback(() => {
-    queryClient.invalidateQueries([ENTITY_STORE_ENGINE_STATUS], {
-      refetchType: 'active',
-    });
-  }, [queryClient]);
-};
-
-export const useInitEntityEngineMutation = (options?: UseMutationOptions<{}>) => {
-  const { telemetry } = useKibana().services;
-  const invalidateEntityEngineStatusQuery = useInvalidateEntityEngineStatusQuery();
-  const { initEntityStore } = useEntityStoreRoutes();
-  return useMutation<InitEntityEngineResponse[]>(
+  return useMutation<InitEntityStoreResponse>(
     () => {
       telemetry?.reportEvent(EntityEventTypes.EntityStoreEnablementToggleClicked, {
         timestamp: new Date().toISOString(),
         action: 'start',
       });
-      return initEntityStore('user').then((usr) =>
-        initEntityStore('host').then((host) => [usr, host])
-      );
+      return enableEntityStore();
     },
     {
+      mutationKey: ENABLE_STORE_STATUS_KEY,
+      onSuccess: () => queryClient.refetchQueries(ENTITY_STORE_STATUS),
       ...options,
-      mutationKey: INIT_ENTITY_ENGINE_STATUS_KEY,
-      onSettled: (...args) => {
-        invalidateEntityEngineStatusQuery();
+    }
+  );
+};
 
-        if (options?.onSettled) {
-          options.onSettled(...args);
-        }
-      },
+export const INIT_ENTITY_ENGINE_STATUS_KEY = ['POST', 'INIT_ENTITY_ENGINE'];
+export const useInitEntityEngineMutation = (options?: UseMutationOptions<{}>) => {
+  const queryClient = useQueryClient();
+
+  const { initEntityEngine } = useEntityStoreRoutes();
+  return useMutation<InitEntityEngineResponse[]>(
+    () => Promise.all([initEntityEngine('user'), initEntityEngine('host')]),
+
+    {
+      mutationKey: INIT_ENTITY_ENGINE_STATUS_KEY,
+      onSuccess: () => queryClient.refetchQueries({ queryKey: ENTITY_STORE_STATUS }),
+      ...options,
     }
   );
 };
 
 export const STOP_ENTITY_ENGINE_STATUS_KEY = ['POST', 'STOP_ENTITY_ENGINE'];
-
 export const useStopEntityEngineMutation = (options?: UseMutationOptions<{}>) => {
   const { telemetry } = useKibana().services;
-  const invalidateEntityEngineStatusQuery = useInvalidateEntityEngineStatusQuery();
-  const { stopEntityStore } = useEntityStoreRoutes();
+  const queryClient = useQueryClient();
+
+  const { stopEntityEngine } = useEntityStoreRoutes();
   return useMutation<StopEntityEngineResponse[]>(
     () => {
       telemetry?.reportEvent(EntityEventTypes.EntityStoreEnablementToggleClicked, {
         timestamp: new Date().toISOString(),
         action: 'stop',
       });
-      return stopEntityStore('user').then((usr) =>
-        stopEntityStore('host').then((host) => [usr, host])
-      );
+      return Promise.all([stopEntityEngine('user'), stopEntityEngine('host')]);
```

Comment:
it might be useful to have an array of supported entity types which we can map over here and below, but maybe we can do this when the service and generic entity are added.